usr/share/rear/skel/default/etc/scripts/unlock-opal-disks

```diff
@@ -201,6 +201,20 @@ while (( attempt < max_authentications )); do
         else
             display_message "$unlocked_device_count of $device_count disks unlocked, rebooting..."
         fi
+        if type -p efibootmgr &>/dev/null; then
+            # Workaround for EFI firmware glitch which prevents booting the real OS after unlocking SEDs.
+            # Symptom: Boot hangs when Grub hands over control to the Linux kernel. Seen on an HPE ML10Gen9 server.
+            # Cf. https://github.com/rear/rear/issues/2425
+            read -r field_name current_boot_number field_junk <<<"$(efibootmgr 2>/dev/null | grep 'BootCurrent:')"
```

Comment:
Only a side note from me who is not at all a UEFI expert
because you grep for `BootCurrent`:
You my have a look at my
https://github.com/rear/rear/issues/2276#issuecomment-559791942
and the subsequent comments from @gozora 
that there are various different ways how the EFI firmware can boot something
so that the `efibootmgr` output could be not reliable - as far as I understand things.
But of course for a workaround for a possinle EFI firmware glitch parsing
the `efibootmgr` output is at least better than nothing.

Reply:
Yes, UEFI firmware can boot even without a proper boot entry present. `RAWDISK` output and other places within ReaR use the default bootloader path, which UEFI systems will discover automatically: https://github.com/rear/rear/blob/fb23c5d711af9ee505a9b03ea7324a098f90891d/usr/share/rear/output/RAWDISK/Linux-i386/270_create_grub2_efi_bootloader.sh#L103-L105

That does not mean that `efibootmgr` output is not reliable. Actually, it is. The tool does nothing but operate on the list of boot entries stored in the machine's NVRAM. `efibootmgr` does the same as your system's firmware setup with regard to boot entries and boot order.

Reply:
O.k. I rephrase it to:
"While the `efibootmgr` output is technically 100% reliable,
its meaning by normal humans interpretation is not."
;-)

Reply:
Well, yes, the entire UEFI thing is not that user friendly. It takes some time to get used to it... ;-)